product_configurator/models/product.py

```diff
@@ -269,6 +269,31 @@ def get_config_image_obj(self, value_ids, size=None):
         return img_obj
 
     @api.multi
+    def encode_custom_values(self, custom_values):
+        """ Hook to alter the values of the custom values before creating or writing
+
+            :param custom_values: dict {product.attribute.id: custom_value}
+
+            :returns: list of custom values compatible with write and create
+         """
+        binary_attribute_ids = self.env['product.attribute'].search([
+            ('custom_type', '=', 'binary')]).ids
```

Comment:
My pedanticism is that a lookup is quicker than an assignment, and so for uses of a looked up value only 2 or 3 times, I often don't bother with assignment, especially if they are not long and wordy, which means assignment can add to readability.

On a different note, though, I notice in v10 Odoo have begun using camel case for these:

`ProductAttribute = self.env['product.attribute']`
as opposed to
`product_attribute_obj = self.env['product.attribute']`

Only if it is not intended to become a record set, but remains an object reference, though.  Not sure if I agree with the change, but it seems to reflect the intention of the assignment, though...